ToyProject/app/src/main/java/com/example/toyproject/ui/main/MainViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.toyproject.ui.main
+
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.toyproject.data.entity.SearchMovieData
+import com.example.toyproject.data.repository.MovieRepositoryImpl
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+
+class MainViewModel() : ViewModel() {
+    val TAG = this.javaClass.name
+    val movieData = MutableLiveData<List<SearchMovieData>>()
+    val query = MutableLiveData<String>()
+    val isLoading = MutableLiveData<Boolean>()
+
+    fun searchMovie() {
+        runBlocking {
+            launch {
+                MovieRepositoryImpl.getMovieData(query.toString(), success = {
+                    movieData.value = it.searchMovieResults
+                }, fail = {
+                    Log.e(TAG, it.message)
+                })
+            }
+            isLoading.value = true
+        }
+        isLoading.value = false
```

Comment:
isLoading.value = true 과 isLoading.value = flase 를 각각 하나의 함수로 만들어서 의미를 명확히 표현하는게 어떨까요?

```suggestion
        showLoading()
        hideLoading()
```

Reply:
https://github.com/StudyFork/CleanArchitectureStudy02/pull/22/commits/3b1843318e244b21b31124015fa575724710545a
늦어서 죄송해요 감사합니다.